server/jetstream_cluster_2_test.go

```diff
@@ -6391,41 +6395,90 @@ func TestJetStreamClusterRePublishUpdateNotSupported(t *testing.T) {
 			if resp.Type != JSApiStreamUpdateResponseType {
 				t.Fatalf("Invalid response type %s expected %s", resp.Type, JSApiStreamUpdateResponseType)
 			}
-			if !IsNatsErr(resp.Error, JSStreamInvalidConfigF) {
-				t.Fatalf("Expected error regarding config error, got %+v", resp.Error)
+			if IsNatsErr(resp.Error, JSStreamInvalidConfigF) {
+				t.Fatalf("Expected no error regarding config error, got %+v", resp.Error)
 			}
 		}
-		expectFailUpdate()
 
-		// Now try with a new stream with RePublish present and then try to change config
-		cfg = &StreamConfig{
-			Name:     stream + "_2",
-			Storage:  MemoryStorage,
-			Replicas: replicas,
-			RePublish: &RePublish{
-				Source:      ">",
-				Destination: "bar.>",
-			},
+		expectRepublished := func(expectedRepub bool) {
+			t.Helper()
+
+			nc, js := jsClientConnect(t, s)
+			defer nc.Close()
+
+			var foos, bars atomic.Int32
+
+			// Create a subscriber for foo.> so that we can see
+			// our published message being echoed back to us.
+			_, err := js.Subscribe(
+				"foo.>",
+				func(msg *nats.Msg) { foos.Add(1) },
+				nats.AckNone(),
+				nats.DeliverNew(),
+			)
+			require_NoError(t, err)
+
+			// Create a subscriber for bar.> so that we can see
+			// any potentially republished messages.
+			_, err = js.Subscribe(
+				"bar.>",
+				func(msg *nats.Msg) { bars.Add(1) },
+				nats.AckNone(),
+				nats.DeliverNew(),
+			)
+			require_NoError(t, err)
+
+			// Publish a message, it will hit the foo.> stream and
+			// may potentially be republished to the bar.> stream.
+			_, err = js.Publish("foo."+stream, []byte("HELLO!"))
+			require_NoError(t, err)
+
+			// Wait for a little while so that we have enough time
+			// to determine whether it's going to arrive on one or
+			// both streams.
+			time.Sleep(time.Millisecond * 100)
+			switch {
+			case foos.Load() != 1:
+				t.Fatalf("Expected 1 message on foo.>, got %d", foos.Load())
+			case expectedRepub && bars.Load() != 1:
+				t.Fatalf("Expected 1 message on bar.>, got %d", bars.Load())
+			default:
+				t.Logf("Got %d on foo.>, %d on bar.>", foos.Load(), bars.Load())
+			}
 		}
-		addStream(t, nc, cfg)
-		cfg.RePublish.HeadersOnly = true
-		expectFailUpdate()
 
-		// One last test with existing first, then trying to remove
-		cfg.Name = stream + "_3"
-		addStream(t, nc, cfg)
+		// At this point there's no republish config, so we should
+		// only receive our published message on foo.>.
+		expectRepublished(false)
+
+		// Add a republish config so that everything on foo.> also
+		// gets republished to bar.>.
+		cfg.RePublish = &RePublish{
+			Source:      "foo.>",
+			Destination: "bar.>",
+		}
+		expectUpdate()
+		expectRepublished(true)
```

Comment:
If we're testing all scenarios, maybe it would be worth it to test changing the existing RePublish into another one?